### PR TITLE
Fix support of Python 3.8

### PR DESCRIPTION
**Pull request by leplatrem**
The parent arguments do not really match what we have in stdlib

https://github.com/python/cpython/blob/v3.8.0/Lib/logging/__init__.py#L413-L421

https://github.com/python/cpython/blob/v3.8.0/Lib/logging/__init__.py#L555-L576

### CODE REVIEW

File: src/dockerflow/logging.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def __init__(self, fmt=None, datefmt=None, style="%", logger_name="Dockerflow"):
         if sys.version_info[:2] < (3, 1):
             parent_init(self, format, datefmt)
         else:
-            parent_init(self, format, datefmt, style)
+            parent_init(self, datefmt=datefmt, style=style)
```
Comment (jezdez) [inline src/dockerflow/logging.py]:
```suggestion
            parent_init(self, fmt=format, datefmt=datefmt, style=style)
```

Comment (leplatrem) [reply src/dockerflow/logging.py]: Unless I misunderstand the stdlib code, I don't really agree with this. 

The `fmt` parameter isn't supposed to be a callable. Look at the default value on the default style https://github.com/python/cpython/blob/v3.8.0/Lib/logging/__init__.py#L413-L421

Comment (leplatrem) [reply src/dockerflow/logging.py]: And BTW what result do you expect by passing the `format` function?

Comment (leplatrem) [reply src/dockerflow/logging.py]: See https://github.com/python/cpython/pull/9703 and https://travis-ci.org/Kinto/kinto/jobs/599165067#L298

**Comment by leplatrem** [comment]
In case this helps, this is the stacktrace of python-dockerflow with Python 3.8:

```
/home/travis/build/Kinto/kinto/.tox/py38/lib/python3.8/site-packages/dockerflow/logging.py:82: in __init__
    parent_init(self, format, datefmt, style)
/opt/python/3.8.0/lib/python3.8/logging/__init__.py:576: in __init__
    self._style.validate()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <logging.PercentStyle object at 0x7f99c30830a0>
    def validate(self):
        """Validate the input format, ensure it matches the correct style"""
>       if not self.validation_pattern.search(self._fmt):
E       TypeError: expected string or bytes-like object
``` 

The `fmt` parameter [should be a string](https://github.com/python/cpython/blob/v3.8.0/Lib/logging/__init__.py#L413-L421). Here, we don't use this style customization and can leave the default (None).

File: src/dockerflow/logging.py (hunk 1 above)
Comment (jezdez) [reply src/dockerflow/logging.py]: Woops, you're right, I meant to pass the `fmt` variable, not the `format` variable.
```suggestion
            parent_init(self, fmt=fmt, datefmt=datefmt, style=style)
```

IIUC we need to keep the call signature of the `logging.Formatter.__init__`, and [it takes a `fmt` parameter](https://github.com/python/cpython/blob/v3.8.0/Lib/logging/__init__.py#L555).

In fact this seems to be a regression introduced in https://github.com/mozilla-services/python-dockerflow/pull/25/files#diff-792fa8a235db26fe95e4b30a7b16b8aeL50-R60! The parameter passed into `__init__` changed from format to fmt, but not the name of the variable passed to the parent init. 😬

File: src/dockerflow/logging.py [hunk 2]
```diff
@@ -79,7 +79,7 @@ def __init__(self, fmt=None, datefmt=None, style="%", logger_name="Dockerflow"):
         if sys.version_info[:2] < (3, 1):
             parent_init(self, format, datefmt)
```
Comment (jezdez) [inline src/dockerflow/logging.py]:
```suggestion
            parent_init(self, fmt, datefmt)
```

**Comment by leplatrem** [comment]
The TravisCI job did not apparently... https://travis-ci.org/mozilla-services/python-dockerflow/pull_requests

File: tox.ini [hunk 3]
```diff
@@ -7,15 +7,16 @@ envlist =
     # the version matrix according to the Django docs:
     # https://docs.djangoproject.com/en/2.0/faq/install/#what-python-version-can-i-use-with-django
     py{27,36}-dj111-tests,
-    py{36,37}-dj{21,22}-tests,
+    py{36,37,38}-dj{21,22}-tests,
     py{27,36}-fl{011,012,10}-tests,
```
Comment (diox) [inline tox.ini]: This should also contain `37` and `38`, right ?

File: tox.ini [hunk 4]
```diff
@@ -7,15 +7,16 @@ envlist =
...
-    py{36,37}-s19-tests,
+    py{36,37,38}-s19-tests,
```
Comment (diox) [inline tox.ini]: Python `3.8` is not in `.travis.yml` so tests aren't run on that version in travis: https://travis-ci.org/mozilla-services/python-dockerflow/builds/602217901

File: tox.ini [hunk 5]
```diff
@@ -7,15 +7,16 @@ envlist =
...
 
 [testenv]
 basepython =
     py27: python2.7
     py36: python3.6
     py37: python3.7
+    py38: python3.8
```
Comment (diox) [inline tox.ini]: Note: maybe we should also consider changing `[testenv:py36-lint]` below.

Comment (diox) [reply tox.ini]: As well as `tests-py{27,34}: PYTEST_ADDOPTS = --ignore=tests/test_sanic.py`.

**Comment by leplatrem** [comment]
Apparently `uvloop` does not like Python 3.8

``` 
   #define CO_OPTIMIZED    0x0001
                           ^
  uvloop/loop.c:334:37: note: in definition of macro ‘__Pyx_PyCode_New’
             PyCode_New(a, 0, k, l, s, f, code, c, n, v, fv, cell, fn, name, fline, lnos)
                                       ^
  uvloop/loop.c:154727:63: note: in expansion of macro ‘CO_OPTIMIZED’
     __pyx_codeobj__13 = (PyObject*)__Pyx_PyCode_New(1, 0, 3, 0, CO_OPTIMIZED|CO_NEWLOCALS, __pyx_empty_bytes, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_tuple__12, __pyx_empty_tuple, __pyx_empty_tuple, __pyx_kp_s_uvloop_loop_pyx, __pyx_n_s_callback, 850, __pyx_empty_bytes); if (unlikely(!__pyx_codeobj__13)) __PYX_ERR(2, 850, __pyx_L1_error)
                                                                 ^
  In file included from /opt/python/3.8.0/include/python3.8/compile.h:5:0,
                   from /opt/python/3.8.0/include/python3.8/Python.h:138,
                   from uvloop/loop.c:20:
  /opt/python/3.8.0/include/python3.8/code.h:122:28: note: expected ‘PyObject * {aka struct _object *}’ but argument is of type ‘int’
   PyAPI_FUNC(PyCodeObject *) PyCode_New(
```

File: tox.ini (hunk 5 above)
Comment (diox) [reply tox.ini]: Ah yeah sorry read that too quickly

**Comment by diox** [comment]
Looks like https://github.com/MagicStack/uvloop/issues/277#issuecomment-545730455

File: tox.ini (hunk 5 above)
Comment (jezdez) [reply tox.ini]: Oh, woops I've been doing those changes locally as well but ran against some issues with the test setup. Will incorporate yours changes

**Comment by jezdez** [comment]
Closing in favor #39.

**Comment by jezdez** [comment]
[dockerflow 2019.10.0](https://pypi.org/project/dockerflow/2019.10.0/) is out!

**Pull request closed without merging**